pkg/cloudprovider/azure.go

```diff
@@ -396,6 +408,20 @@ func (a *Azure) getAuthorizer(env azureapi.Environment, clientID, clientSecret,
 	return authorizer, nil
 }
 
+// getNodeLock retrieves node lock from nodeLockMap, If lock doesn't exist, then update map
+// with a new lock entry for the given node name.
+func (a *Azure) getNodeLock(nodeName string) *sync.Mutex {
+	var nodeLock *sync.Mutex
+	var ok bool
+	a.nodeMapLock.Lock()
+	if nodeLock, ok = a.nodeLockMap[nodeName]; !ok {
+		nodeLock = &sync.Mutex{}
+		a.nodeLockMap[nodeName] = nodeLock
+	}
+	a.nodeMapLock.Unlock()
+	return nodeLock
```

Comment:
I think returning nodeLock here is wrong as it will always return a new pointer, and not one to the node?

I think this here is easier to read:
~~~
func (a *Azure) getNodeLock(nodeName string) *sync.Mutex {
	a.nodeMapLock.Lock()
        defer a.nodeMapLock.Unlock()
	if _, ok := a.nodeLockMap[nodeName]; !ok {
		a.nodeLockMap[nodeName] = &sync.Mutex{}
	}
	return a.nodeLockMap[nodeName]
}
~~~

Reply:
yes, done.